lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1166,8 +1166,7 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
         }
         assert(isa<CaseStmt>(initialCaseVarDecl->getParentPatternStmt()));
 
-        if (vd->getInterfaceType() && initialCaseVarDecl->getType() &&
```

Comment:
#VarDecl::getType() also computes the interface type.

Reply:
Ah nice, even more cleanup